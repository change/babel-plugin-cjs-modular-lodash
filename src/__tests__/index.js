const fs = require('fs');
const path = require('path');
const { transform } = require('babel-core');
const plugin = require('../index');

function transformFixture(fixturePath, options) {
  return transform(
    fs.readFileSync(path.join(__dirname, '../../support/fixtures', fixturePath)),
    Object.assign(
      {
        babelrc: false,
        plugins: [[plugin, {}]],
      },
      options
    )
  ).code;
}

test('Transforms calls on global _ properties to modular requires', () => {
  expect(transformFixture('global.js')).toMatchSnapshot();
});

test('Transforms methods in files with a monolithic require and removes monolithic require', () => {
  expect(transformFixture('monolithic_require.js')).toMatchSnapshot();
});

test('Transforms deconstructured assignment correctly', () => {
  expect(transformFixture('deconstruction.js', { presets: [['es2015', { modules: false }]] })).toMatchSnapshot();
});
