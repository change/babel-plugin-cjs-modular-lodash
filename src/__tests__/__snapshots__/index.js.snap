// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transforms calls on global _ properties to modular requires 1`] = `
"const _lodash_map = require(\\"lodash/map\\");

const _lodash_identity = require(\\"lodash/identity\\");

const _lodash_isNumber = require(\\"lodash/isNumber\\");

_lodash_map([], x => x);
_lodash_map([], x => !!x);
_lodash_identity;
_lodash_isNumber(0);"
`;

exports[`Transforms methods in files with a monolithic require and removes monolithic require 1`] = `
"const _lodash_map = require('lodash/map');

const _lodash_identity = require('lodash/identity');

const _lodash_isNumber = require('lodash/isNumber');

_lodash_map([], x => x);
_lodash_map([], x => !!x);
_lodash_identity;
_lodash_isNumber(0);"
`;
